way_to_home/notification/views.py

```diff
@@ -1 +1,127 @@
-"""This module that provides base logic for CRUD of notification`s model objects."""
+"""
+Notification view module
+================
+
+This module that provides base logic for CRUD of notification`s model objects.
+"""
+
+from django.views import View
+from django.http import HttpResponse, JsonResponse
+
+from notification.models import Notification
+from way.models import Way
+
+
+class NotificationView(View):
+    """
+    Notification view that handles GET, POST, PUT, DELETE requests and provides
+    appropriate operations with notification model.
+    """
+    def get(self, request, way_id, notification_id=None):
+        """ Method that handles GET request. """
+        user = request.user
+        way = Way.get_by_id(way_id)
+
+        if not way:
+            return HttpResponse('way is not found', status=400)
+
+        if way.user != user:
+            return HttpResponse('permission denied', status=403)
+
+        if not notification_id:
+            data = [notification.to_dict() for notification in way.notifications.all()]
+            return JsonResponse(data, status=200, safe=False)
+
+        notification = Notification.get_by_id(notification_id)
+        if not notification:
+            return HttpResponse('Notification not found', status=400)
+
+        if notification.way != way:
+            return HttpResponse('permission denied', status=403)
+
+        return JsonResponse(notification.to_dict(), status=200)
+
+    def put(self, request, way_id, notification_id=None):  # pylint: disable=R0201
+        """ Method that handles PUT request. """
+        user = request.user
+        data = request.body
+        way = Way.get_by_id(obj_id=way_id)
+
+        if not notification_id:
+            return HttpResponse('obj_id are not received', status=400)
+
+        notification = Notification.get_by_id(obj_id=notification_id)
+
+        if not (notification or way):
+            return HttpResponse('failed, obj not found', status=400)
+
+        if not (way == notification.way or user == way.user):
+            return HttpResponse('permission denied', status=403)
```

Comment:
The same situation as mentioned above. If change step of retrieving `notification`, you don't need to check this: 
```way == notification.way```

Reply:
done